e2e/__tests__/failureDetailsProperty.test.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import * as path from 'path';
+import {isJestCircusRun} from '@jest/test-utils';
+import runJest from '../runJest';
+
+const cleanAbsolutePaths = (stdout: string) =>
+  stdout.replace(
+    new RegExp(path.resolve(__dirname, '../..'), 'gu'),
+    '<rootDir>',
+  );
+
+test('that the failureDetails property is set', () => {
+  const {stdout} = runJest('failureDetails-property', ['tests.test.js']);
+
+  if (isJestCircusRun()) {
+    expect(cleanAbsolutePaths(stdout)).toMatchInlineSnapshot(`
+      "Custom reporter output:
+      [
+        '[\\\\n' +
```

Comment:
It would be cool if we could remove some of these slashes somehow, but both `JSON.stringify` and `util.inspect` give roughly this output (main difference is that `util.inspect` doesn't have the stack trace on the single line, but I believe that's required for CI as it was failing on that before for some reason).

Reply:
could we slice out the "Custom reporter output:" part in the test and just do `JSON.parse`?

Reply:
Good thinking - if we just call `JSON.stringify` on the whole array & log that rather than logging the stringify version of each failureDetail, we half the number of slashes, but that then lets us call `JSON.parse` on `stdout` resulting in a very nice snapshot <3